### ISSUE TITLE
Calculate_Itemized_Deductions is performed by nothing — every return takes the standard deduction

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#1161]
Found triaging the negative-gap scenario cohort (#1140 burn-down): scenarios
where the rules compute MORE tax than expected, by scenario-specific amounts.

## The bug

`Calculate_Itemized_Deductions` — which performs the medical / SALT /
mortgage-interest / charitable subtree — is **performed by zero tables**. So
`result.total_itemized` is always 0, and `Calculate_Deductions`' chooser
(`total_itemized > standard_deduction`) always takes standard. Every itemizing
taxpayer in the corpus is over-taxed. Verified on
`TestCase_ScheduleA_04_Charitable_Mixed`: expected deduction 38,000 (15,000
mortgage interest + 12,000 property tax + charitable), rules gave the 31,500
MFJ standard; the audit trail contains no itemization lines at all.

The orphan-call analysis flags calls to *undefined* tables, not defined tables
nobody calls — which is how this survived every gate. (A "defined but
unreachable from any entry table" advisory would have caught it; worth
considering alongside the #776 reachability work.)

## The fix, verified

One line, through the API:

```
{"op":"add-initial-action","dsl":"perform Calculate_Itemized_Deductions"}
```

on `Calculate_Deductions` (its initial actions currently perform only
`Calculate_Standard_Deduction`).

## Why it is not merged

Applied, the scenario ratchet moves 280 → 277: the itemized cohort comes
clean, but a handful of previously-clean scenarios carried standard-based
expectations despite itemizable inputs — their expecteds must be reconciled
(gap should equal itemized − standard exactly, same principled substitution as
the #1159 pass). The ratchet refusing a net regression is it working as
designed. The reconciliation is one focused session: apply the one-liner,
identify the flipped scenarios, verify each gap is exactly itemized−standard,
update, raise the floor past 280.

**Comment by PaulSnow** [comment]
Did the fix and measured it end to end. **The orphan is real and the one-liner is necessary — but it is not sufficient, and my original claim that it repairs the itemized cohort is wrong.** Correcting that here with the measurements.

## What the fix actually does

Applied `{"op":"add-initial-action","dsl":"perform Calculate_Itemized_Deductions"}` to `Calculate_Deductions`, rebuilt, and probed all 505 scenarios before and after:

| | before | after |
|---|---|---|
| clean | 280 | 277 |
| **newly clean** | | **0** |
| newly failing | | 3 |
| outputs changed at all | | 10 |

**Zero scenarios are repaired.** The ScheduleA cohort I cited moves in the right direction but stops short — `ScheduleA_02` 218,500 → 210,000 against an expected 213,000 (now over-deducting by 3,000); `ScheduleA_03` 118,500 → 114,500 against 112,000 (still under by 2,500); `ScheduleA_01` (medical) does not move at all. So the itemized subtree has component defects independent of the orphan.

## The chooser is verified correct

All three newly-failing scenarios have a gap that is *exactly* itemized − standard:

- `AdditionalTaxes_11` — itemized 50,000, Single std 15,750, gap **34,250** ✓
- `State_WA_01` — itemized 60,500, Single std 15,750, gap **44,750** ✓
- `TestCase_Family_2025` — itemized 33,000, MFJ std 31,500, gap **1,500** ✓

`Calculate_Deductions`' larger-of test is right. These three were clean only because they were never itemizing.

## Why I did not adopt those three computed values

The #1159 substitution rule requires the new value be derivable from the scenario's own declared inputs. It is not, and the reason is a second defect:

**`Sum_SALT_Deduction` iterates a `deductions` array, and scenario property data does not fully populate it.** `TestCase_Family_2025` declares two properties (taxes 8,500 and 4,200) and SALT collected is **8,500** — only the first. Likewise `State_WA_01`'s itemized 60,500 has ~23,000 I cannot account for from its visible inputs (mortgage 28,000 + SALT 9,500).

So the itemized *totals* are not yet trustworthy, and writing them into expected values would bake unverified numbers into the corpus's baseline. Reverted; the ratchet holds at 280.

## What this issue now needs (in order)

1. **Medical**: `Calculate_Medical_Expense_Deduction` contributes nothing even when expenses exceed the 7.5%-of-AGI floor (`ScheduleA_01` unchanged by the fix).
2. **SALT collection**: property/state-tax data from scenarios must reach the `deductions` array — check the mapping and `Sum_SALT_Deduction`'s source, per-property rather than first-only.
3. **Mortgage / charitable**: residual ±2,500–3,000 in `ScheduleA_02/03` after the above.
4. **Then** apply the one-liner, reconcile the (by then verifiable) flipped scenarios, and raise the floor past 280 in one coherent step.

The one-liner is correct and should ship — as the *last* step of that sequence, not the first, because alone it trades three verified-clean scenarios for zero repairs.
